mapper/mt101_utilities.bal

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/financial.iso20022.payment_initiation as painIsoRecord;
+import ballerinax/financial.swift.mt as swiftmt;
+
+# Get the instructing party from the Pain001 document.
+#
+# + document - The Pain001 document
+# + return - The instructing party or an empty record
+isolated function getMT101InstructingPartyFromPain001Document(painIsoRecord:Pain001Document document)
+returns swiftmt:MT50C?|swiftmt:MT50L? {
+    painIsoRecord:Party52Choice? id = document.CstmrCdtTrfInitn.GrpHdr.InitgPty.Id;
+
+    if (id is ()) {
+        return ();
+    }
+
+    painIsoRecord:OrganisationIdentification39? OrgId = id?.OrgId;
+    if (OrgId != ()) {
+        return <swiftmt:MT50C>{
+            name: MT50C_NAME,
+            IdnCd: {
+                content: getEmptyStrIfNull(OrgId?.AnyBIC),
+                number: NUMBER1
+            }
+        };
+    }
```

Comment:
Do we need seperate utility classes. Can't we make them common?

Reply:
For ISO documents, they can have different record types. Thus so we can't use common methods for some mappings.